### ISSUE TITLE
Rule Change: add support for sorting keys of function parameters of type object

### ISSUE THREAD

**Issue by HonzaMac** [eslint/eslint#18029]
### What rule do you want to change?

sort-keys

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new default behavior

### Example code
[Playground](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IHNvcnQta2V5czogW1wiZXJyb3JcIiwgXCJhc2NcIl0gKi9cbi8qZXNsaW50LWVudiBlczYqL1xuY29uc3QgY29tcG9uZW50cyA9ICh7YiwgYX0pID0+IHtcbiAgY29uc29sZS5sb2coJ05PT1AnKTtcbn0iLCJvcHRpb25zIjp7ImVudiI6eyJlczIwMjEiOnRydWV9LCJydWxlcyI6eyJzb3J0LWtleXMiOlsiZXJyb3IiXX0sInBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fSwiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoibW9kdWxlIn19fQ==)

```js
/* eslint sort-keys: ["error", "asc"] */
/*eslint-env es6*/
const components = ({b, a}) => {
  console.log('NOOP');
}
```

### What does the rule currently do for this code?

It does nothing. Parameter sorting is ignored.

### What will the rule do after it's changed?

It will sort object parameter keys.
```
/* eslint sort-keys: ["error", "asc"] */
/*eslint-env es6*/
const components = ({a, b}) => {
  console.log('NOOP');
}
```

### Participation

- [X] I am willing to submit a pull request to implement this change.
